source/src/plugins/KolmogorovTest.cc

```diff
@@ -0,0 +1,171 @@
+/// \file DQMQualityTest.cc
+/*
+ *
+ * DQMQualityTest.cc source template automatically generated by a class generator
+ * Creation date : ven. f�vr. 20 2015
+ *
+ * This file is part of DQM4HEP libraries.
+ *
+ * DQM4HEP is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * based upon these libraries are permitted. Any copy of these libraries
+ * must include this copyright notice.
+ *
+ * DQM4HEP is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with DQM4HEP.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @author Remi Ete
+ * @copyright CNRS , IPNL
+ */
+
+// -- dqm4hep headers
+#include <dqm4hep/Internal.h>
+#include <dqm4hep/Logging.h>
+#include <dqm4hep/MonitorElement.h>
+#include <dqm4hep/PluginManager.h>
+#include <dqm4hep/QualityTest.h>
+#include <dqm4hep/StatusCodes.h>
+#include <dqm4hep/XmlHelper.h>
+#include <dqm4hep/AnalysisHelper.h>
+
+// -- root headers
+#include <TH1.h>
+#include <TMath.h>
+
+namespace dqm4hep {
+
+  namespace core {
+
+    /** KolmogorovTest class
+     */
+    class KolmogorovTest : public QualityTest {
+    public:
+      class Factory : public QualityTestFactory {
+      public:
+        QTestPtr createQualityTest(const std::string &qname) const override;
+      };
+
+      KolmogorovTest(const std::string &name);
+      ~KolmogorovTest() override = default;
+      StatusCode readSettings(const dqm4hep::core::TiXmlHandle xmlHandle) override;
+      std::string getTestOptions(const bool isHistogram, bool m_useUnderflow, bool m_useOverflow);
```

Comment:
m_useUnderflow and m_useOverflow are member variables, you don't need to pass them as argument